addOns/jwt/src/main/java/org/zaproxy/zap/extension/jwt/JWTActiveScanner.java

```diff
@@ -0,0 +1,242 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.jwt;
+
+import java.io.IOException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.jwt.attacks.ClientSideAttack;
+import org.zaproxy.zap.extension.jwt.attacks.ServerSideAttack;
+import org.zaproxy.zap.extension.jwt.exception.JWTException;
+import org.zaproxy.zap.extension.jwt.utils.JWTUtils;
+
+/**
+ * JWT Scanner is used to find the vulnerabilities in JWT implementations. <br>
+ * Resources containing more information about vulnerabilities in implementations are: <br>
+ *
+ * <ol>
+ *   <li><a href="https://tools.ietf.org/html/draft-ietf-oauth-jwt-bcp-06" >JSON Web Token Best
+ *       Practices(IETF document)</a>
+ *   <li><a
+ *       href="https://cheatsheetseries.owasp.org/cheatsheets/JSON_Web_Token_Cheat_Sheet_for_Java.html">
+ *       OWASP cheatsheet for vulnerabilities in JWT implementation</a>
+ *   <li><a href="https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries">For
+ *       server side vulnerabilities in JWT implementations</a>
+ * </ol>
+ *
+ * @author KSASAN preetkaran20@gmail.com
+ * @since TODO add version
+ */
+public class JWTActiveScanner extends AbstractAppParamPlugin {
+
+    private static final int PLUGIN_ID = 1001;
+    private static final String NAME = JWTI18n.getMessage("ascanrules.jwt.name");
+    private static final String DESCRIPTION = JWTI18n.getMessage("ascanrules.jwt.description");
+    private static final String SOLUTION = JWTI18n.getMessage("jwt.scanner.soln");
+    private static final String REFERENCE = JWTI18n.getMessage("jwt.scanner.refs");
+    private static final Logger LOGGER = Logger.getLogger(JWTActiveScanner.class);
+    private int maxRequestCount;
+
+    public JWTActiveScanner() {}
+
+    @Override
+    public void init() {
+        switch (this.getAttackStrength()) {
+            case LOW:
+                maxRequestCount = 4;
+                break;
+            case MEDIUM:
+                maxRequestCount = 8;
+                break;
+            case HIGH:
+                maxRequestCount = 12;
+                break;
+            case INSANE:
+                maxRequestCount = 28;
+                break;
+            default:
+                maxRequestCount = 8;
+                break;
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        String newValue = value.trim();
+        newValue = JWTUtils.extractingJWTFromParamValue(newValue);
+
+        if (!JWTUtils.isTokenValid(newValue)) {
+            LOGGER.info("Token: " + newValue + " is not a valid JWT token.");
```

Comment:
Reduce this one as well (debug?), might end up flooding the logs as most of them will not be JWT tokens.